### PR TITLE
Implement CheckmarkableChipAttributes on ChoiceChip

### LINKED ISSUE #124662: When using Material 3, the ChoiceChip check mark cannot be hidden.
I have looked into the implementation of ChoiceChip, and it appears that it does not implement CheckmarkableChipAttributes like FilterChip does.

## Steps to Reproduce

1. Set the userMaterial3 property of ThemeData to true.
2. To add the ChoiceChip widget to the project
3.  ChoiceChip widget does not have a showCheckmark property, so the check mark in the widget will always be displayed.

**Expected results:** 
screenshot2

**Actual results:** 
screenshot2

<details>
<summary>Code sample</summary>

```dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter/material.dart';

/// Flutter code sample for [ActionChoice].

void main() => runApp(const ChipApp());

class ChipApp extends StatelessWidget {
  const ChipApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(colorSchemeSeed: const Color(0xff6750a4), useMaterial3: true),
      home: const ActionChoiceExample(),
    );
  }
}

class ActionChoiceExample extends StatefulWidget {
  const ActionChoiceExample({super.key});

  @override
  State<ActionChoiceExample> createState() => _ActionChoiceExampleState();
}

class _ActionChoiceExampleState extends State<ActionChoiceExample> {
  int? _value = 1;

  @override
  Widget build(BuildContext context) {
    final TextTheme textTheme = Theme.of(context).textTheme;

    return Scaffold(
      appBar: AppBar(
        title: const Text('ActionChoice Sample'),
      ),
      body: Center(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Text('Choose an item', style: textTheme.labelLarge),
            const SizedBox(height: 10.0),
            Wrap(
              spacing: 5.0,
              children: List<Widget>.generate(
                3,
                (int index) {
                  return ChoiceChip(
                    label: Text('Item $index'),
                    // showCheckmark: false,
                    selected: _value == index,
                    onSelected: (bool selected) {
                      setState(() {
                        _value = selected ? index : null;
                      });
                    },
                  );
                },
              ).toList(),
            ),
          ],
        ),
      ),
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
```

```
```

```
```

</details>

### PR DESCRIPTION
**Pull request by 9oya**
I believe that there is a mistake in the implementation of the ChoiceChip widget as it does not have the CheckmarkableChipAttributes property like the FilterChip and InputChip widgets. To address this issue, I have added the implementation of CheckmarkableChipAttributes to ChoiceChip like the other widgets. Additionally, I have created issue #124662

fixes #124662

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by 9oya** [comment]
@TahaTesser @huycozy @GayathrideviGA  Please review this [PR](https://github.com/flutter/flutter/pull/124743) before merge this [PR](https://github.com/flutter/flutter/pull/124779)

File: packages/flutter/lib/src/material/choice_chip.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/choice_chip.dart:168]:
```suggestion
      showCheckmark: showCheckmark ?? chipTheme.showCheckmark ?? Theme.of(context).useMaterial3,
```


Please add `chipTheme.showCheckmark`.

Since we're defaulting here instead of leaving it for `RawChip` to handle. We should have added the theme value before the default value.

File: packages/flutter/test/material/choice_chip_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/choice_chip_test.dart:148]: This doesn't look like it belongs in this PR. 
The PR's for `showCheckmark` isn't it?

File: packages/flutter/test/material/choice_chip_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/choice_chip_test.dart:136]:
```suggestion
  testWidgets('ChoiceChip passes checkmark properties to RawChip', (WidgetTester tester) async {
```

Test both `showCheckmark` and `checkmarkColor` properties are passed to `RawChip`

File: packages/flutter/test/material/choice_chip_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/choice_chip_test.dart:146]:
```suggestion
    const Color checkmarkColor = Color(0xff0000ff);
    await tester.pumpWidget(wrapForChip(
      child: const ChoiceChip(
        label: Text('Test'),
        selected: true,
        showCheckmark: showCheckmark,
        checkmarkColor: checkmarkColor,
    )));
    final RawChip rawChip = tester.widget(find.byType(RawChip));
    expect(rawChip.showCheckmark, showCheckmark);
    expect(rawChip.checkmarkColor, checkmarkColor);
  });
```

File: packages/flutter/lib/src/material/choice_chip.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/choice_chip.dart:169]:
```suggestion
      checkmarkColor: checkmarkColor,
      tooltip: tooltip,
```

Pass the `ChoiceChip.checkmarkColor` to the `RawChip` you added. 
We don't need to pass the theme value here since `RawChip` will handle it.

File: packages/flutter/test/material/choice_chip_test.dart
Comment (TahaTesser) [reply packages/flutter/test/material/choice_chip_test.dart:136]: We should add the same test but use `ChipTheme` to pass these properties.

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/lib/src/material/choice_chip.dart
Comment (9oya) [reply packages/flutter/lib/src/material/choice_chip.dart:168]: You're right. I'll add chipTheme.showCheckmark before the default Theme.of(context).useMaterial3.

File: packages/flutter/test/material/choice_chip_test.dart
Comment (9oya) [reply packages/flutter/test/material/choice_chip_test.dart:148]: I referred to this test case in the `FilterChip` widget tests. You're right, this unit test is not directly related to this PR. I'll remove it.

File: packages/flutter/test/material/choice_chip_test.dart
Comment (TahaTesser) [reply packages/flutter/test/material/choice_chip_test.dart]: @9oya  
Can you please duplicate this test and use `ChipTheme` to test if `showCheckmark` is passed from the theme?

Comment (9oya) [reply packages/flutter/test/material/choice_chip_test.dart]: Okay, I will 🙆‍♂️

Comment (9oya) [reply packages/flutter/test/material/choice_chip_test.dart]: @TahaTesser I am trying to check `ChipTheme`, but it isn't affected by `ChoiceChip's` property like `RawChip` does. You can check `class _ChoiceChipDefaultsM3 extends ChipThemeData` in `choice_chip.dart`. 
The `checkmarkColor` and `showCheckmark` properties of `ThemeData(useMaterial3: true).chipTheme` will always return null values.

Comment (TahaTesser) [reply packages/flutter/test/material/choice_chip_test.dart]: We've tested `ChoiceChip` properties which you updated here. Just do the same for theme value we're passing

Add this test below the `ChoiceChip passes iconTheme property to RawChip'` test.


```dart
  testWidgets('ChoiceChip passes showCheckmark from ChipTheme to RawChip', (WidgetTester tester) async {
    const bool showCheckmark = false;
    const Color checkmarkColor = Color(0xff0000ff);
    await tester.pumpWidget(wrapForChip(
      child: const ChipTheme(
        data: ChipThemeData(
          showCheckmark: showCheckmark,
          checkmarkColor: checkmarkColor,
        ),
        child: ChoiceChip(
          label: Text('Test'),
          selected: true,
          showCheckmark: showCheckmark,
        ),
      )));
    final RawChip rawChip = tester.widget(find.byType(RawChip));
    expect(rawChip.showCheckmark, showCheckmark);
  });
```

Comment (TahaTesser) [reply packages/flutter/test/material/choice_chip_test.dart]: Screenshot 2023-04-19 at 15 08 21

Comment (TahaTesser) [reply packages/flutter/test/material/choice_chip_test.dart]: We don't need to test the `checkmarkColor` theme value here since we're not providing it here.

Comment (9oya) [reply packages/flutter/test/material/choice_chip_test.dart]: Oh now I got your point!

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by TahaTesser** [comment]
Need a second review from @HansMuller to land this.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by MoyinShabi** [comment]
@9oya Hi, thanks a lot for this fix! But when do I get to use the `showCheckMark` property of the M3 `ChoiceChip` widget, please? I'm on the flutter sdk version 3.10.5 and the property still doesn't exist for the widget...

**Comment by TahaTesser** [comment]
> @9oya Hi, thanks a lot for this fix! But when do I get to use the `showCheckMark` property of the M3 `ChoiceChip` widget, please? I'm on the flutter sdk version 3.10.5 and the property still doesn't exist for the widget...

Hi! This is currently available on the `beta` channel.  It should land in the `stable` channel during the next major release.


You can see a list of Flutter version tags assigned to this commit by clicking the commit above.

Screenshot 2023-07-14 at 11 07 46


Screenshot 2023-07-14 at 11 06 19

**Comment by MoyinShabi** [comment]
Awesome, thank you!!

**Comment by 9oya** [comment]
Thanks @TahaTesser :)

**Pull request merged**
